File: Python3/1149.py
n,m = map(int,input().split())
goods ={}
for _ in range(n):
    a,b = map(int,input().split())
    if goods.get(a) is None:
        goods[a] = []
    if goods.get(b) is None:
        goods[b] = []
    goods[a].append(b)
    goods[b].append(a)
for _ in range(m):
    __,*cur_goods = map(int,input().split())
    exists = {x:True for x in cur_goods}
    flag = False
    for good in cur_goods:
        if goods.get(good) is None: continue
        for each in goods[good]:
            if exists.get(each) is not None:
                flag = True
                break
        if flag:    break
    if flag:
        print("No")
    else:
        print("Yes")

File: c++/1149.cpp
//
//  main.cpp
//  test
//
//  Created by 李寻欢 on 2019/10/19.
//  Copyright © 2019 李寻欢. All rights reserved.
//
#include <stdio.h>
#include <iostream>
#include <string>
#include <map>
#include <cmath>
#include <algorithm>
#include <stack>
#include <vector>
#include <queue>
#include <unordered_map>
#include <unordered_set>
#include <string.h>
#include <set>
#include <deque>
using namespace std;

int n,m,k,temp;
unordered_map<int, vector<int>> goods;

int main(){
    scanf("%d%d",&n,&m);
    int a,b;
    for(int i=0;i<n;++i){
        scanf("%d%d",&a,&b);
        goods[a].emplace_back(b);
        goods[b].emplace_back(a);
    }
    while (m--) {
        scanf("%d",&k);
        vector<int> v;
        bool exists[100000] = {false};
        for(int i=0;i<k;++i){
            scanf("%d",&temp);
            exists[temp] = true;
            v.emplace_back(temp);
        }
        bool flag = true;
        for(int i=0;i<k && flag;++i){
            for(auto &each:goods[v[i]]){
                if(exists[each]){
                    flag = false;
                    break;
                }
            }
        }
        printf("%s\n",flag==true?"Yes":"No");
    }
}

